app/validator.js

```diff
@@ -27,6 +27,14 @@ const customJoi = Joi.extend((joi) => ({
     ]
 }));
 
+const vizIdPrefix='threshold_alert_:';
+
+function preprocessor(result){
+	result.value.report_line_item.visualizationid=result.value.report.thresholdAlert?vizIdPrefix+result.value.report_line_item.visualizationid:result.value.report_line_item.visualizationid;
```

Comment:
what is the code convention that we are using? I have never seen that there is no whitespace used between variables or operators. https://blog.risingstack.com/javascript-clean-coding-best-practices-node-js-at-scale/

It's much readable with whitespaces, but if we chose without, just let me know so we would stick to the same style